Source/Core/Core/HW/DVD/DVDThread.cpp

```diff
@@ -358,6 +363,37 @@ static void FinishRead(u64 id, s64 cycles_late)
                                        buffer);
 }
 
+static void PatchReadRequest(std::vector<u8> &buffer, u64 offset) {
+  if(!s_disc_patches)
+    return;
+
+  u64 end_offset = offset + buffer.size();
+
+  // Scan through patches which might overlap our current read request and apply them
+  for (auto it = s_disc_patches->begin(); it != s_disc_patches->end(); it++) {
+    if (it->first > end_offset) // Patch starts after the end of the read request
+    {
+      return;
+    }
+    else if (it->first >= offset) // Patch starts inside the read request
+    {
+      u64 start = it->first - offset;
+      u32 old_data;
+      u32 new_data;
+      std::memcpy(&old_data, buffer.data() + start, 4);
```

Comment:
```suggestion
      std::memcpy(&old_data, buffer.data() + start, sizeof(old_data));
```

ditto for the memcpy below this one, but utilizing new_data instead